aliyun-ecs/src/main/java/org/jclouds/aliyun/ecs/handlers/ECSRetryableErrorHandler.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.aliyun.ecs.handlers;
+
+import com.google.common.annotations.Beta;
+import org.jclouds.aliyun.ecs.domain.ErrorMessage;
+import org.jclouds.http.HttpCommand;
+import org.jclouds.http.HttpResponse;
+import org.jclouds.http.functions.ParseJson;
+import org.jclouds.http.handlers.BackoffLimitedRetryHandler;
+import org.jclouds.logging.Logger;
+
+import javax.annotation.Resource;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * This handles failed responses that return a <code>DependencyViolation</code>.
+ * <p>
+ * In order to determine if the error is retryable, the response body must be
+ * read, so this handler will have to buffer the response payload in memory so
+ * the response body can be read again in subsequent steps of the response
+ * processing flow.
+ */
+@Singleton
+@Beta
+public class ECSRetryableErrorHandler extends BackoffLimitedRetryHandler {
+
+   private static final String RETRYABLE_ERROR_CODE = "DependencyViolation";
+
+   @Resource
+   protected Logger logger = Logger.NULL;
+   private final ParseJson<ErrorMessage> parseError;
+
+   @Inject
+   ECSRetryableErrorHandler(ParseJson<ErrorMessage> parseError) {
+      this.parseError = parseError;
+   }
+
+   @Override
+   public boolean shouldRetryRequest(HttpCommand command, HttpResponse response) {
+      // Only consider retryable errors and discard rate limit ones
+      if (response.getStatusCode() != 403) {
+         return false;
```

Comment:
You should call `super` here. This class just handles 403 errors; otherwise fallback to the default class that handles retries.